### PR TITLE
Demolish smarter

### PR DESCRIPTION
**Pull request by klikooo**
With this we first demolish the largest piece of land by a catergory, in this case the gardens.

User story:
As a develop, I want my virtual human to have enough space to build on so that it can improve its student housing indicator.

### CODE REVIEW

**Comment by MaxGroenenboom** [comment]
I get an exception when I try to run it: `java.lang.Exception: found errors while parsing: [Validation error at line 6, position 9 in actions.act2g: 'planned(Id,Level,MultiPolygon)' is not defined as either knowledge or a belief].
    at goal.tools.eclipse.DebugTool.main(DebugTool.java:64)`

~~I don't know why, it fails on things that haven't been changed in a long while.~~
Found the problem, it uses a new predicates file, but it doesn't use a new action spec file, so it still tries to verify all actions in that new action spec while only the last two actions are used.

**Comment by klikooo** [comment]
That is weird since it does not fail here locally, will try to fix it.

**Comment by MaxGroenenboom** [comment]
Fixed by updating GOAL, it works now and does what's expected. I approve.

**Comment by MarcianoJ** [comment]
Works as expected. Maybe self/1 and maxArea/3 should be in predicates.pl instead of predicatesDemolish.pl

File: src/DUWOagent/demolishPredicates.pl [hunk 1]
```diff
@@ -0,0 +1,30 @@
+[predicates].
+ 
+:- dynamic getAreaOwnBuilding/2.
+:- dynamic maxArea/3.
+:- dynamic buildingAreas/2.
+:- dynamic demolished/1.
+:- dynamic self/1.
+:- dynamic buildings/1.
+ 
+ % Get a buildings id and area by catergory.
```
Comment (MaxGroenenboom) [inline src/DUWOagent/demolishPredicates.pl]: typo `catergory`, should be `category`

**Comment by klikooo** [comment]
Self/1 is needed because if it is not defined it will return a parse exception.

**Comment by MaxGroenenboom** [comment]
It works for me now, approved (again :stuck_out_tongue:)

**Comment by MarcianoJ** [comment]
I approve, too!

**Pull request merged**